pkg/controller/grouppermission/grouppermission_controller.go

```diff
@@ -53,6 +54,13 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// Watch for changes to secondary resource Namespace
```

Comment:
Can you expand on the "why" behind the Watch? Readers will be able to see that there's a watch, but we can set the context for them to have a sentence or two about the "why." Eg "Watch Namespaces because we need to .... on create or deletion"